### ISSUE TITLE
[New build - DEFAULT] 20w21c: Teleport Station, new Starting Zone.

### ISSUE THREAD

**Issue by tsunamayo** [tsunamayo/Starship-EVO#2121]
The Teleport station block has been added. 
In survival this block will be needed to teleport back to your ship!
VFX and SFX to come.

Changes and new features:
- Teleport Station added.
- Starting Zone updated and tweaked (Credit @ Womble)

Bug Fixes:
#2111 Supercharge addon counter incorrect.
#2038 Player see equiped yellow box on startup on AMD card.
Invalid turret link on startup.
Ship anti-gravity not activated on control.
#2115 Hinge hitbox issue.
Lag when entering ship command.
Holo preview on item stand visible through walls.

**Comment by tsunamayo** [comment]
Build is live! Thanks for testing, this will be the build for the upcoming wave!

**Comment by Sullos** [comment]
Love the build, everything working well so far on my end. Question regarding the new 'Tank' block... are we readily able to start incorporating the block into our builds, does it serve a purpose or should we just ignore it?
Thanks Tsuna!

**Comment by Kaiser-Indrasil** [comment]
I'd like to share some concerns on the teleportation station.

I love the idea of having to build a boarding/docking ramp the crew would have to cross to get onto a different station/ship. The teleporter basically renders these concepts obsolete.
I fear the thought that in order to board hostile ships, all we'd need to do is to teleport onto them once they get close enough instead of having to somehow anchor our ship onto theirs.

Still, the idea to "fast-travel" is sound but I feel like it needs to be limited. Maybe limit the teleportation to the habitat only?
Or, perhaps even better, change its behavior to a respawning station? That way, you could still be able to fast-travel to your ship, but you'd lose everything your character has in their personal inventory in the process, a bit like in Minecraft or Star Citizen.
What about traveling to big ships that cannot land on the habitat/planets? The solution comes in the form of planetary shuttles you use to travel planetside and which you can land in your mothership's hangar/dock with it externally.

**Comment by tsunamayo** [comment]
@Sullos I forgot to hide it, it was still wip... I will push a new build real quick.
Yes teleport will be restricted, but I need it for boarding typically. If you want to board manually by breaching good to you, but I will never to manage to make this work with AI crews.
Same for the docking ramp, I will try but the AI will be quite complicated.
There will be a respawn station in the form of a med bay, but the behaviour will be bit different: you only respawn to your last visited medbay.

**Comment by Womble-TC** [comment]
What about having the ability to summon your ship like in no man's sky. You can press a button, and your ship makes its way towards you and lands. I can imagine it might be a pain to code, but it might be more interesting that having the player just teleport.

**Comment by tsunamayo** [comment]
@Womble-TC then how you board? I like the idea of ship summoning though. They are not exclusive, say you just found a rich ore on a planet, if you want to mine ect you can ask your ship to come over.

**Comment by ProPeach** [comment]
Most people design boarding ramps with a button to activate them on the outside, or they could use the new volumetric trigger to open up their ship. A summoning option would be utterly awesome.

**Comment by Garrett-C** [comment]
> Most people design boarding ramps with a button to activate them on the outside, or they could use the new volumetric trigger to open up their ship. A summoning option would be utterly awesome.

I feel like he means how do you board as in how would summoning help when boarding enemy ships. But I could be wrong.

**Comment by ZachZent** [comment]
I am not the biggest fan of the teleporter, but in this case I guess it makes sense. So next question, how exactly will this work in gameplay. If you can only teleport from tele-block to block, then you or the other ship can just delete it and never be at risk from boarding. If you can teleport anywhere on the ship, that would be very cheap and how would you pick what spot you board given ships are custom?

**Comment by ZachZent** [comment]
There is also the airlock docking talked about awhile ago. That would connect the two ships in a more "determined" way

**Comment by Kaiser-Indrasil** [comment]
Okay, I came up with the idea with the teleportation boarding in mind.
How about, if the boarding ship forced a teleportation link to a hostile ship, it would have to:
1. Stay in close proximity or else the link is severed
2. Spend lots of energy to maintain the forced teleportation link, probably forcing it to power some of its systems down (weapons for example).
That way they would have to disable or cripple the target ship first, before attempting to board it, or else it would leave them vulnerable to a counter-attack.

**Comment by Garrett-C** [comment]
> Okay, I came up with the idea with the teleportation boarding in mind.
> How about, if the boarding ship forced a teleportation link to a hostile ship, it would have to:
> 
> 1. Stay in close proximity or else the link is severed
> 2. Spend lots of energy to maintain the forced teleportation link, probably forcing it to power some of its systems down (weapons for example).
> 
> That way they would have to disable or cripple the target ship first, before attempting to board it, or else it would leave them vulnerable to a counter-attack.

You could probably also add that maybe the shields have to be down on the enemy ship, or maybe there is a teleportation jammer that has to be taken offline.

**Comment by Womble-TC** [comment]
@Garrett-C could have something like in xwing where you can "disable" a ships systems by shooting it with enough Ion weapons. Once the countermeasures are down, the enemy can board using the telporters.

**Comment by Kaiser-Indrasil** [comment]
@Womble-TC Yeah but that doesn't make it risky for the boarders. I still think that forcing a teleportation link to a hostile ship should draw a lot of power.

**Comment by Garrett-C** [comment]
> @Garrett-C could have something like in xwing where you can "disable" a ships systems by shooting it with enough Ion weapons. Once the countermeasures are down, the enemy can board using the telporters.

Yeah Ion weapons would be a good addition and could work well to achieve that goal. Funny you should mention the Xwing games, I played in if them fairly recently.

**Comment by Garrett-C** [comment]
> @Womble-TC Yeah but that doesn't make it risky for the boarders. I still think that forcing a teleportation link to a hostile ship should draw a lot of power.

Not really sure about a sustained powerdraw honestly. I see why you are suggesting it but surely if all of your ai crew are off of your ship then it doesn’t matter what the power issue is you are still vulnerable.

**Comment by Womble-TC** [comment]
@Kaiser-Indrasil Oh, I totally agree with that power drain, just thought of something else that could make it more interesting too

**Comment by Womble-TC** [comment]
One last thing, what's to stop a player just building a slider door around the teleporter so no enemies can get out? *open the blast doors, open the blast doors!*

**Comment by Creepthefrenchpastery** [comment]
Or they could just remove / shut off the teleporter.

**Comment by Dwarf-LordPangolin** [comment]
I'll wait to give more feedback on the whole teleporting thing until later, but for now:

Tsuna, you'd asked previously if we had any suggestions for blocks that should be placed on the brick grid; if it's possible, could the teleporter block be one of those? It currently moves on the block grid, and integrating these is going to be a bit tricky with that.

**Comment by tsunamayo** [comment]
@Dwarf-LordPangolin yes I will change it it is a miss from my side.
Is it too tall? The collision grid is a box for now but I could improve it.
Now of course you wont spawn on an enemy respawn! Although now you can, actually I added that now so I can debug my NPC spawning routine ;). The normal spawn function spawn you out of the ship so it is not good for that purpose.
You will have to lock on to the enemy ship, and it would spawn at a targeted location inside the ship, more or less close to it depending on the level of your crew operator. It will require quite a lot of power. I dont think it would be too op, I mean if it we will just balance it, there will be many ways.
I realized I still havent shared my plan for crew, as this is 100% a crew thing.

**Comment by tsunamayo** [comment]
Oh one of of the natural defense against crew boarding as that you will have medbay with respawn. So the defending ship is respawning close the combat while for the attacking attrition will make it harder.

**Comment by ZachZent** [comment]
Two things. A concern and an idea.

Concern: I'm not too sure about the medbay for respawning NPCs. Now for players absolutely, but for NPCs I don't see what benefit it will add except make boarding actions take that much longer. The numbers on each side will be the same, skill on each side would be the same. You could add balancers like medical supplies, but in the end it just makes boarding actions take longer. Could stop some people from trying to board as it becomes more of a slog. Maybe instead of respawning, medics who are based in the medbay can heal NPC/players. 

NPCs are a resource, you have what you have and once they are gone they are gone. You have to get more. It is a risk for the attackers, they can't just keep coming and coming. Same with using NPCs for other tasks like mining. They are not an infinite resource just like everything else.

Idea: When the first NPCs teleport over, it creates an "exit" pad. All boarders will exit though this pad. The defenders goal is to destroy this pad which will stop the transports. The attackers have whatever goal they came to do, sabotage, steal something, or just kill everyone.

Though the NPCs are limited to transporting, the player isn't. Say someone breaches though the wall. What does it accomplish, the player is just inside the ship. To give the player more options and to make boarding more tactical, players can place secondary teleporter exits anywhere else on the ship allowing NPCs to transport to the secondary pad(s) instead of the primary one. For NPC boarders, for example pirates, they would be able to place these down as well though probably in less tactically advantageous positions as they are only NPCs. The defenders can take out these secondary pads stopping the transports. If the primary one is taken out, all secondary ones stop working.

For balance, the energy cost to transport over to these secondary pads would require the energy to transport to the primary plus extra depending on the distance. Or just a flat rate for simplicity. Crafting these should be semi-expensive. That way most pirates can have say one or two, but lots to capture a massive capital ship would be very expensive.

**Comment by tsunamayo** [comment]
So sorry I havent shared my doc but there is too much discussion at the moment, I cant follow all at once, will give it in a few days.
Now the things for crew is they will be level and rarity based, so I cant get them killed too easily. I want to make a fun game, not a game too punishing. They will level-up, gain better stats, and once a high level a special capacity will be unlocked. Rarer crew would more powerful.

But I dig your exit pad idea!
 Honestly crew boarding will be highly experimental, I will get a first version up and running and we will build from there.
I am not too worried about balancing, you always have many options to boost or nerf something. 
If med-bay are too powerful then we will nerf them. What if your med bay cost a shit load of power and takes 5 min to regen a guy? I bet you might prefer more firepower instead. Med bay is just an option to make the game deeper (do you destroy a reactor first or a medbay) and will give a way to balance the whole thing.

**Comment by Dwarf-LordPangolin** [comment]
> Is it too tall? The collision grid is a box for now but I could improve it.
Just a bit; a 1/8 or 1/16 meter height would probably be better. Stepping off it currently results in a bit of a jolt! XD

**Comment by ZachZent** [comment]
Fair enough. I still dislike the idea of a medbay, but why not take a page from FTL's book. Well another page (FTL is the best game, well second best :D).

> Each time a crew member is resurrected, (s)he loses a portion (12.5%-15% depending on skill) of the skill experience. 

This added with regeneration time and other balances should be enough. And perhaps taking another page from FTL, have the crew regeneration be a clone bay. When it gets destroyed, the dead crew will all die permanently

**Comment by tsunamayo** [comment]
yeah FTL is a great inspiration to me. We will see for the detail, but all of that is definitely possible, if we want to add a bit more punishment.

**Comment by Kaiser-Indrasil** [comment]
Taking a page from FTL's book, medbays in that game heal the crew over time so you gotta choose - either heal up wounded crewmates or subject them to a no-retreat defensive, risking their death while increasing your chances at repeling the boarders. Yes, I love FTL, too.

**Comment by Garrett-C** [comment]
Yeah FTL is fine for what it is. When getting a new ships is as easy as pressing new game it’s not a massive issue. But in Skywanderers where loosing your ship will have a significant cost, either based on time or the in game credits, you don’t want it to be too punishing. 

Obviously you don’t want it to be too easy but permadeath or massive crew penalties sounds a bit too unfun.

Honestly I am pretty meh about crew boarding in general. I mean it should definitely be a thing but I don’t think it should be the main thing for combat. Personally boarding should really only be a thing at the end of the battle. I don’t like the idea of just teleporting onto a ship at the start of the battle and winning that way.

**Comment by tsunamayo** [comment]
I think the important is to give player options, some will do that, some wont. I will have to be careful at making it balanced, so it is not the only way to do, but it is not useless either.
Plus I could make some interesting missions out of that: infiltrate a ship, steal something, deliver hostage, whatnot. But I am getting ahead of myself. The difference between all the other spacesim out there is that we just _simply dont play a ship_. In other spacesim, you are the ship. I think it would be a shame to not leverage that aspect of the game.
Yes in FLT you loose your ship often, but honestly I expect it would be also the case here. There will be penalty of some sort, but I want people to go to fight, try to pick enemy maybe a bit too strong, just have fun and experiment really. 
Remember when I asked people about sorting what they want in game? Fun was always on top!
The main difference with FLT I would say is FTL was really tactical, you controlled each crew precisely. Here is it is gonna be more messy, you have to give up on the strategic view and play the fun card instead - I am totally fine with that. I wont have time to go Operation Flashpoint style with precise order anyway.
Thanks

**Comment by Garrett-C** [comment]
My only concern with the idea of loosing your ship easily is how that will effect things in the early game.

When you have enough resources to build or buy a new ship loosing one isn’t a massive issue but if you have just built your first ship and managed to fly of loosing your ship could be very annoying. Loosing everything you have spent hours building up is never fun, loosing it before you have a chance to build up more stuff is even worse.

Like let’s say you start on a planet and take a few hours to be able to build a ship. You then fly off and loose your ship in say half an hour or so. Now you need to spend a few hours getting a new ship. That’s going to suck. I don’t think having punishing mechanics is bad, I do think it would be bad to lose evening so quickly.

**Comment by ZachZent** [comment]
I agree that can be an issue and a complete wipe is very disheartening, but there are ways around it that still has risk. One is that NPCs will level up in their early levels faster and higher levels slower. That way even if you lose a crew member, you won't be stuck with a level 1 for the rest of time. A second is that NPCs crew, at least those who are base levels compared to higher level crew which would cost more to hire, is a inexpensive resource. Maybe the real cost is keeping them alive, say with food. For a small ship that would be cheap and you would be able to relatively easily keep your crew happy and healthy. As the ships gets larger and more crew needed, the cost to keep them alive increases quite a lot. A fair balance and in a way a soft size limiter

**Comment by Garrett-C** [comment]
> I agree that can be an issue and a complete wipe is very disheartening, but there are ways around it that still has risk. One is that NPCs will level up in their early levels faster and higher levels slower. That way even if you lose a crew member, you won't be stuck with a level 1 for the rest of time. A second is that NPCs crew, at least those who are base levels compared to higher level crew which would cost more to hire, is a inexpensive resource. Maybe the real cost is keeping them alive, say with food. For a small ship that would be cheap and you would be able to relatively easily keep your crew happy and healthy. As the ships gets larger and more crew needed, the cost to keep them alive increases quite a lot. A fair balance and in a way a soft size limiter

I have to say I think keeping them alive is better. The ability to loose your whole crew to a pirate boarding just sounds potentially horrible. It means any small glitch in AI or pathfinding could ruin a fight. I don’t think AI permadeath is the way to go. 

Additionally if permadeath was a thing for the AI then it would encourage people to build big in order to have as many AI as possible, I don’t see that being good for any servers.

**Comment by tsunamayo** [comment]
No guys my intent was not clear: I want people loose their ship with a very small respawn cost. Aka go have fun blow your ship, it is insured no pb. You will need to pay some cash for each block you place, it is like an insurance cost. Then once you respawn you will have a small cash penalty and else (like you loose what you won since the last shipyard land).
Same with crew, they die you respawn them no pb. Maybe add a happy meter with some malus, then a crew that just die wont be much happy for a while. Too much unhappy maybe a mutinery? There is so much things we can do.
Yes I want an upkeep cost for crew, but maybe in the form of a % of winnings.

**Comment by Garrett-C** [comment]
Interesting, I am still concerned that a new player could possibly get sent back to square 1 with that system but it’s more of something that would need to be worked out with that once things like survival are being looked into more.

Still not sure about crew members being fully killed like that. I mean one fear is that it turns combat into a game of who has the bigger crew. Which doesn’t sound ideal from a fun or performance standpoint.

**Comment by Garrett-C** [comment]
That said it arguably becomes a war of attrition either way. If crews can respawn then it gives smaller ships a chance but given that larger crews would also respawn it might not make a massive difference. 

Ultimately boarding could easily come down to big crew wins.

**Comment by ZachZent** [comment]
>  mean one fear is that it turns combat into a game of who has the bigger crew.

I mean that is kind of the point. Well that plus skill of boarders. It shouldn't always be a fair fight. Now if it was player count I would agree, but since it is NPCs you can always get more. Hopefully there will be an option to surrender to save your crew.

> If crews can respawn then it gives smaller ships a chance but given that larger crews would also respawn it might not make a massive difference.

That is why I'm a bit weary about crew respawning in your ship. Not just for the defenders, but because the attackers can just respawn without risk to their medbay. Even if you add constraints it won't stop a player from placing a lot of them getting a large amount of crew at once even if it takes a bit longer. Maybe crew can only be respawned at stations with a Clone Corp Enterprise stall or something.

**Comment by Garrett-C** [comment]
> > mean one fear is that it turns combat into a game of who has the bigger crew.
> 
> I mean that is kind of the point. Well that plus skill of boarders. It shouldn't always be a fair fight. Now if it was player count I would agree, but since it is NPCs you can always get more. Hopefully there will be an option to surrender to save your crew.
> 
> > If crews can respawn then it gives smaller ships a chance but given that larger crews would also respawn it might not make a massive difference.
> 
> That is why I'm a bit weary about crew respawning in your ship. Not just for the defenders, but because the attackers can just respawn without risk to their medbay. Even if you add constraints it won't stop a player from placing a lot of them getting a large amount of crew at once even if it takes a bit longer. Maybe crew can only be respawned at stations with a Clone Corp Enterprise stall or something.

Yeah I totally expect that a 2 man shuttle should not be able to win against a fully maned 500m ship or something. But if it becomes a game of “how many AI can I stuff onto my ship” it runs the risk of ruining everything else in combat. 

The issue with crews not respawning is how much of a pain it will be if a crew member you invest hours into getting to a high level is killed because of something simple and unfair like a bug are a pathfinding error. I don’t know how many people will find it fun to loose crew, but I would imagine most people would be unhappy to loose crew as a result of something completely out of their control. It honestly sounds like a recipe for disaster if you ask me.

**Comment by tsunamayo** [comment]
You wont "loose" a crew, your crew will die, and you will respawn them. It is okay if the player loose combat from time to time. We are getting too far on this, could you guys wait for the crew doc first, I have too much brainstorm topic opened atm ;)

**Comment by Garrett-C** [comment]
> You wont "loose" a crew, your crew will die, and you will respawn them. It is okay if the player loose combat from time to time. We are getting too far on this, could you guys wait for the crew doc first, I have too much brainstorm topic opened atm ;)

Lol that's fair. Yeah it's perma death for crew that I meant. Will be interested to see your crew documents down the line to see what ideas you have. Will be easier to discuss then.
